packages/svelte/src/internal/client/render.js

```diff
@@ -1225,14 +1232,26 @@ export function bind_prop(props, prop, value) {
 /**
  * @param {Element} element_or_component
  * @param {(value: unknown) => void} update
+ * @param {import('./types.js').MaybeSignal} binding
  * @returns {void}
  */
-export function bind_this(element_or_component, update) {
+export function bind_this(element_or_component, update, binding) {
 	untrack(() => {
+		// If a bind_this is shared across multple targets, then we
+		// need to ensure we correctly assign the value rather than
+		// only setting `null` for when they unmount.
+		if (is_signal(binding)) {
+			if (bind_this_context === null) {
+				bind_this_context = [];
```

Comment:
Will this work correctly across multiple components that use `bind:this`?
Can we somehow avoid the global? For example by doing something like "wait one tick after the render effect is destroyed (below in line 1252), and if the binding now is different to my instance then don't set it to `null`"

Reply:
I tried that, that approach didn't work. This does work across multiple components too :)